BUILD.md

```diff
@@ -0,0 +1,132 @@
+# ParFlow Build System
```

Comment:
I agree, we currently have a first step but we should definitely consolidate based on the "reference usage/build". This can definitely be achieved via later PRs, but the "knowledge of usage" should help shape the way we want the documentation to looks like. Also we tried to create side directories/docs to streamline the first integration by limiting as much as possible any impact on core Parflow while creating content that could then be moved/integrated later while requiring less effort.